botorch/optim/optimize.py

```diff
@@ -257,22 +270,49 @@ def _optimize_acqf_batch(opt_inputs: OptimizeAcqfInputs) -> tuple[Tensor, Tensor
 
     initial_conditions_provided = opt_inputs.batch_initial_conditions is not None
 
+    required_raw_samples = opt_inputs.raw_samples
     if initial_conditions_provided:
-        batch_initial_conditions = opt_inputs.batch_initial_conditions
+        provided_initial_conditions = opt_inputs.batch_initial_conditions
+        if opt_inputs.raw_samples is not None:
+            required_raw_samples -= provided_initial_conditions.shape[-2]
```

Comment:
seems that this would error out if `opt_inputs.raw_samples` were `None` (which seems possible)?

Reply:
I don't think so as the logic is based on `required_raw_samples` below which is assigned in 269. I think that the check in 272 should be `if required_raw_samples is not None:` and then it's easier to follow